tsconfig.es5.json

```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./tsconfig.base.json",
   "compilerOptions": {
-    "target": "es5"
+    "target": "es5",
+    "downlevelIteration": true
```

Comment:
Though, Map/Set is available since ES2015. TypeScript doesn't introduce any polyfills for Map/Set. I doubt that the es5-esm products we are publishing right now do not actually work for es5 environments.

The good news is that the es5-esm products do not contain any fashion syntaxes (other than the ESM part) so that end-users do not need to transpile the otel-packages. They only need to include the polyfill packages. I believe this doesn't break the assumptions https://github.com/open-telemetry/opentelemetry-js/pull/2472#issuecomment-919921603.